infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/StreamsSnapshotWriter.java

```diff
@@ -241,64 +241,44 @@ public void apply(LogReplicationEntryMsg message) {
             return;
         }
         UUID regularStreamId = opaqueEntry.getEntries().keySet().stream().findFirst().get();
-
-
-        // When an upgrade has been performed and Active's streams to replicate set has streams not
-        // present in Standby's set, they will be dropped for version compatibility consideration
-        if (!regularToShadowStreamId.containsKey(regularStreamId)) {
-            log.warn("Stream {} sent from Active is not expected in Standby. LR could be undergoing a rolling upgrade", regularStreamId);
-            recvSeq++;
-            return;
-        }
-
-        // Clear regular stream on-demand (i.e., as streams come) and only on the first occurrence
+        UUID shadowStreamId = getShadowStreamUUID(regularStreamId);
+        regularToShadowStreamId.put(regularStreamId, shadowStreamId);
+        log.trace("Shadow stream=[{}] for regular stream=[{}]", shadowStreamId, regularStreamId);
+
+        // TODO: how to drop ACTIVE only streams in rolling upgrade case
+//
+//        // When an upgrade has been performed and Active's streams to replicate set has streams not
+//        // present in Standby's set, they will be dropped for version compatibility consideration
+//        if (!regularToShadowStreamId.containsKey(regularStreamId)) {
+//            log.warn("Stream {} sent from Active is not expected in Standby. LR could be undergoing a rolling upgrade", regularStreamId);
+//            recvSeq++;
+//            return;
+//        }
+
+        // Collect the streams that has evidenced data from active. These will be cleared during the apply phase.
         if (!replicatedStreamIds.contains(regularStreamId)) {
             // Note: we should not clear the shadow stream as this could overwrite our mergeOnlyStreams when
             // shadow stream is applied to the regular stream. Shadow streams are seeked on each replication cycle
             // and are GC'ed by the checkpoint / trim.
-            clearRegularStream(regularStreamId);
```

Comment:
this was added as part of https://github.com/CorfuDB/CorfuDB/pull/3132.  We must clear the data of any stream which was written to on the active.